opentimelineio/opentime.py

```diff
@@ -533,27 +573,52 @@ def from_timecode(timecode_str, rate):
 
     :return: (:class:`RationalTime`) Instance for the timecode provided.
     """
+    # Validate rate
+    validate_timecode_rate(rate)
+
+    # Check if rate is drop frame
+    rate_is_dropframe = rate in VALID_DROPFRAME_TIMECODE_RATES
 
+    # Check if timecode indicates drop frame
     if ';' in timecode_str:
-        raise ValueError('Drop-Frame timecodes not supported.')
+        if not rate_is_dropframe:
+            raise ValueError(
+                'Timecode "{}" indicates drop-frame rate.'
```

Comment:
...maybe including the fact that you're detecting the `;` in the string as the reason for why its going down this path?